modules/core/src/convert.cpp

```diff
@@ -5371,6 +5371,52 @@ static bool ocl_convertScaleAbs( InputArray _src, OutputArray _dst, double alpha
     return k.run(2, globalsize, NULL, false);
 }
 
+static bool ocl_convertFp16( InputArray _src, OutputArray _dst )
+{
+    int type = _src.type(), depth = CV_MAT_DEPTH(type), cn = CV_MAT_CN(type);
+
+    int nSChannels = _src.getMat().channels();
+    int nSDims = _src.getMat().dims;
+
+    int ddepth = 0;
+
+    switch( depth )
+    {
+    case CV_32F:
+        ddepth = CV_16S;
+        break;
+    case CV_16S:
+        ddepth = CV_32F;
+        break;
+    default:
+        CV_Error(Error::StsUnsupportedFormat, "Unsupported input depth");
+        return false;
+    }
+
+    _dst.create( nSDims, _src.getMat().size, CV_MAKETYPE(ddepth, nSChannels) );
+    int kercn = 1;
+    int rowsPerWI = 1;
+    String build_opt = format("-D HALF_SUPPORT -D dstT=%s -D srcT=%s -D rowsPerWI=%d%s",
+                           ddepth == CV_16S ? "half" : "float",
+                           ddepth == CV_16S ? "float" : "half",
+                           rowsPerWI,
+                           ddepth == CV_16S ? " -D FLOAT_TO_HALF " : "");
+    ocl::Kernel k("convertFp16", ocl::core::halfconvert_oclsrc, build_opt);
+    if (k.empty())
+        return false;
+
+    UMat src = _src.getUMat();
+    UMat dst = _dst.getUMat();
```

Comment:
I'm not sure about this point.
The ```ReadOnlyNoSize``` and ```WriteOnly``` function follows after here, so I'm not sure if ```ACCESS_READ``` and ```ACCESS_WRITE``` is necessary here.
If ```_src``` or ```_dst``` is already ```UMat```, this ```getUMat(ACCESS_READ)``` would try to grab a row of the image out of the range.

I believe it's OK to use the default ```-1``` for this function.
How is that ?